src/Options/OptionsDetails.hpp

```diff
@@ -42,6 +42,43 @@ std::string name() noexcept {
   return name_helper<T>::name();
 }
 
+// Traverses the group hierarchy of `T`, returning the topmost group that is
+// a subgroup of `Root`. Directly returns `T` if it has no group.
+// This means that the returned type is always the direct child node of `Root`
+// that contains `T` in its hierarchy.
+// If `Root` is not in the group hierarchy of `T`, this function returns the
+// topmost group of `T` (meaning that `Root` is treated as the root of its
+// group hierarchy).
+template <typename T, typename Root, typename = cpp17::void_t<>>
+struct find_subgroup {
+  using type = T;
+};
+
+template <typename T>
+struct find_subgroup<T, typename T::group, cpp17::void_t<typename T::group>> {
+  using type = T;
+};
+
+template <typename T, typename Root>
+struct find_subgroup<T, Root, cpp17::void_t<typename T::group>> {
+  using type = typename find_subgroup<typename T::group, Root>::type;
+};
+
+template <typename T, typename Group, typename = cpp17::void_t<>>
```

Comment:
You should document `is_in_group` since you also documented `find_subgroup`

Reply:
Not sure what I can add to the documentation that isn’t said by the name :D

Reply:
Added something anyway